### PR TITLE
proposing changes in SinOsc

### PR DESCRIPTION
**Pull request by tiagmoraismorgado**
hey guys. i have basically updated the reviews i had previously done to supercollider sinosc documentation
i would like to know wether this can be useful for the community. kind regards

### CODE REVIEW

File: HelpSource/Classes/SinOsc.schelp [hunk 1]
```diff
@@ -13,51 +13,404 @@ Technically, code::SinOsc:: uses the same implementation as  link::Classes/Osc::
...
+ ## link::Classes/SinOscFB:: – sinewave with phase feedback
+ ## link::Classes/PMOsc:: – phase modulation sine oscillator
+ ## link::Classes/Klang:: – bank of sinewave oscillators
+ ## link::Classes/DynKlang:: – modulable bank of sinewave oscillators
+ ::
+
+This is the same as  link::Classes/Osc::  except that the table is
+a sine table of 8192 entries.
+
+From Osc Supercollider Documentation Description:
```
Comment (snappizz) [inline HelpSource/Classes/SinOsc.schelp]: I don't see the point in copy-pasting the Osc docs. They're already linked and this information doesn't seem relevant to people using SinOsc.

File: HelpSource/Classes/SinOsc.schelp [hunk 2]
```diff
@@ -13,51 +13,404 @@ Technically, code::SinOsc:: uses the same implementation as  link::Classes/Osc::
...
+
+This can be acheived by creating a Buffer object and sending it one of the "b_gen" messages ( Buffer: -sine1, Buffer: -sine2, Buffer: -sine3 ) with the wavetable flag set to true.
+
+This can also be achieved by creating a Signal object and sending it the 'asWavetable' message, thereby creating a Wavetable object in the required format. Then, the wavetable data may be transmitted to the server using the Buffer: *sendCollection or Buffer: *loadCollection methods.
 
+////////////////////////////////
+////////////////////////////////
+////////////////////////////////
+
+About wavetable synthesis:
```
Comment (snappizz) [inline HelpSource/Classes/SinOsc.schelp]: This is tangential. SinOsc *internally* uses wavetable synthesis for efficiency, but I definitely wouldn't categorize SinOsc as a wavetable synthesis ugen.

File: HelpSource/Classes/SinOsc.schelp [hunk 3]
```diff
@@ -13,51 +13,404 @@ Technically, code::SinOsc:: uses the same implementation as  link::Classes/Osc::
...
+////////////////////////////////
+////////////////////////////////
 
 classmethods::
 
 method::ar, kr
 
 argument::freq
 Frequency in Hertz.
-Sampled at audio-rate.
```
Comment (snappizz) [inline HelpSource/Classes/SinOsc.schelp]: What happened to this line?

File: HelpSource/Classes/SinOsc.schelp [hunk 4]
```diff
@@ -13,51 +13,404 @@ Technically, code::SinOsc:: uses the same implementation as  link::Classes/Osc::
...
 argument::add
 This value will be added to the output.
 
-
 Examples::
 
 code::
 
+//////////////////////////////////////////
+//////////////////////////////////////////
```
Comment (snappizz) [inline HelpSource/Classes/SinOsc.schelp]: No need for these separators.

**Comment by tiagmoraismorgado** [comment]
i wrote a couple of changes to the entire thing, trying to catalyse your feedback. please check if this is what you mean. kind regards

**Comment by tiagmoraismorgado** [comment]
i tried to fix the feedback everyone gave me but i had an error in the branch, namely in the compilation. can someone help me to fix this? cheers

File: HelpSource/Classes/SinOsc.schelp [hunk 5]
```diff
@@ -314,53 +314,53 @@ Synth.new(\RmOctaver) ;
...
+	if(args[1] == 60, {~midiSinB.value; args.postln;});
+	if(args[1] == 61, {~midiSinB.value; ~midiSinC1.value; args.postln;});
+	if(args[1] == 62, {~midiSinB.value; ~midiSinC2.value; args.postln;});
+	if(args[1] == 63, {~midiSinB.value; ~midiSinC3.value; args.postln;});
+	if(args[1] == 64, {~midiSinB.value; ~midiSinC4.value; args.postln;});
+	if(args[1] == 65, {~midiSinB.value; ~midiSinC5.value; args.postln;});
+	if(args[1] == 66, {~midiSinB.value; ~midiSinC6.value; args.postln;});
+	if(args[1] == 67, {~midiSinB.value; ~midiSinC7.value; args.postln;});
+	if(args[1] == 68, {~midiSinB.value; ~midiSinC8.value; args.postln;});
+	if(args[1] == 69, {~midiSinB.value; ~midiSinC9.value; args.postln;});
```
Comment (snappizz) [inline HelpSource/Classes/SinOsc.schelp]: that's a lot of repeated code -- use arrays!

Comment (vivid-synth) [inline HelpSource/Classes/SinOsc.schelp]: I haven't read the rest of this patch, but you might also want to use switch statements - check out the "Control Structures" help file

**Comment by telephon** [comment]
the SinOsc helpfile isn't a good place for this. Please have a look at the MIDI help files and see if they explain what you found missing.
